SignalServiceKit/src/Messages/TSMessagesManager.m

```diff
@@ -447,7 +447,7 @@ - (void)handlePreKeyBundleAsync:(OWSSignalServiceProtosEnvelope *)preKeyEnvelope
             NSData *plaintextData;
             @try {
                 // Check whether we need to refresh our PreKeys every time we receive a PreKeyWhisperMessage.
-                [TSPreKeyManager checkPreKeys];
+                [TSPreKeyManager didReceivePreKeyWhisperMessage];
```

Comment:
The time window is probably sufficiently small for almost all users, but I'm wondering about public signal numbers (e.g. hotlines) where it's likely they get a deluge of messages from first-time-senders in a short time. What if instead we parameterized this with the PreKeyWhisperMessage on line 452, like this:

```
PreKeyWhisperMessage *message = [[PreKeyWhisperMessage alloc] initWithData:encryptedData];
[TSPreKeyManager didReceivePreKeyWhisperMessageWithPrekeyID:message.preKeyId];
```

Then PreKeyManager could keep an in-memory cache to check only once per preKeyId (per launch), rather than dealing with timeouts.

One complication, we want to make sure we **always** check prekey count when receiving a message without a  one time PreKey. Though that's probably true of the existing PR too.